tests/run/quote-splice-interpret-1.check

```diff
@@ -0,0 +1,2 @@
+true
+false
```

Comment:
How can I be sure the output is from compile-time, not run-time?

Reply:
Because the `inline def` has a top level `~` it can always be inlined. Then the inlined tree is interpreted to evaluate the `Expr[T]` which is then spliced in (see #3853 for a bit of info on that regard).
In short a macro call is guaranteed to be inlined or the compilation fails.